core/org.openjdk.jmc.flightrecorder.rules.jdk/src/main/resources/org/openjdk/jmc/flightrecorder/rules/jdk/messages/internal/messages.properties

```diff
@@ -35,25 +35,25 @@ Preference_SHORT_RECORDING_LONG=Minimum length of a recording to be considered u
+AllocationByThreadRule_TEXT_MESSAGE=The most allocations were likely done by thread ''{0}'' at: {1}
+AllocationRuleFactory_TEXT_CLASS_INFO_LONG=Frequently allocated types are good places to start when trying to reduce garbage collections. Look at where the most common types are being allocated to see if many instances are created along the same call path. Try to reduce the number of instances created by invoking the most commonly taken paths less.
+AllocationRuleFactory_TEXT_THREAD_INFO_LONG=Many allocations performed by the same thread might indicate a problem in a multi-threaded program. Look at the stack traces for the thread with the highest allocation rate. See if the allocation rate can be brought down, or balanced among the active threads.
 ApplicationHaltsRule_HALTS_INFO_LIMIT=Application halts info limit
 ApplicationHaltsRule_HALTS_INFO_LIMIT_DESC=Ratio between application halts and execution time within a time window needed to trigger an info notice
 ApplicationHaltsRule_HALTS_WARNING_LIMIT=Application halts warning limit
 ApplicationHaltsRule_HALTS_WARNING_LIMIT_DESC=Ratio between application halts and execution time within a time window needed to trigger a warning
 ApplicationHaltsRule_HALTS_WINDOW_SIZE=Application halts time window size
 ApplicationHaltsRule_HALTS_WINDOW_SIZE_DESC=The time window size used when evaluating the rule
 # {0} is a concatenated string of event type ids
-ApplicationHaltsRule_EXTRA_EVENT_TYPES=Enabling the following event types would improve the accuracy of this rule: {0}.
+ApplicationHaltsRule_EXTRA_EVENT_TYPES=Enabling the following event types would improve the accuracy of this rule: {0}
 ApplicationHaltsRule_RULE_NAME=Application Halts
-ApplicationHaltsRule_RULE_TEXT=Application efficiency was affected by halts.
+ApplicationHaltsRule_RULE_TEXT=Application efficiency was affected by halts
 # {0} is a percentage, {1} is a timespan, {2} is a timestamp, {3} is a percentage
-ApplicationHaltsRule_RULE_TEXT_LONG=The highest ratio between application halts and execution time was {0} % during {1} starting at {2}. {3} % of the halts during the timespan were because of other reasons than GCs.<p>The total halts ratio during the entire recording was {4} %. {5} % of the total halts were because of other reasons than GCs.
-ApplicationHaltsRule_RULE_TEXT_RECOMMENDATION=Application halts are often caused by garbage collections, but can also be caused by for example excessive thread dumps or heap dumps. Investigate the VM Operation information and possibly the safepoint specific information.
+ApplicationHaltsRule_RULE_TEXT_LONG=The highest ratio of application halts to execution time was {0}% for {1} at {2}. {3}% of the halts were for reasons other than GC.<p>The halts ratio for the entire recording was {4}%. {5}% of the total halts were for reasons other than GC.</p>
```

Comment:
We try to keep a space between value and unit. See https://physics.nist.gov/cuu/Units/checklist.html. See also ISO 31-0.